07_array_algos/practice/solutions.cpp

```diff
@@ -0,0 +1,357 @@
+
+    size_t counter = 0;
+    size_t i = 0;
+    size_t j = 0;
+
+    while (i < size && j < size) {
+        if (first_arr[i] <= second_arr[j])
+            result_arr[counter++] = first_arr[i++];
+        
+        else
+            result_arr[counter++] = second_arr[j++];
+    }
+
+    while(i < size)
+        result_arr[counter++] = first_arr[i++];
+
+    while (j < size) 
+        result_arr[counter++] = second_arr[j++];
+}
+
+// task_07
+void sort_matrix(int matrix[][MAX_COLS], size_t rows, size_t cols) {
+    for (size_t i = 0; i < rows; i++) {
+        sort_arr(matrix[i], cols);
+    }
+
+    for (size_t i = 0; i < cols; i++) {
+        for (size_t j = 0; j < rows; j++) {
+            size_t min_index = j;
+            int min_val = matrix[j][i];
+
+            for (size_t k = j + 1; k < rows; k++) {
+                if (min_val > matrix[k][i]) {
+                    min_val = matrix[k][i];
+                    min_index = k;
+                }
+            }
+            
+            if (min_index != j)
+                std::swap(matrix[min_index][i], matrix[j][i]);
+        }
+    }
+}
+
+// helper function for task_08
+int get_min(int arr[], size_t size) {
+
+    int min_elem = arr[0];
+    size_t min_ind = 0;
+
+    for (size_t i = 0; i < size; i++) {
+        if(min_elem > arr[i]) { 
+            min_elem = arr[i];
+            min_ind = i;
+        }
+    }
+    
+    std::swap(arr[min_ind], arr[size - 1]);
+
+    return min_elem;
+}
+
+void print_array(int arr[], size_t size) {
+
+    for (size_t i = 0; i < size; i++) {
+        std::cout << arr[i] << ' ';
+    }
+    std::cout << '\n';
+}
+
+// task_08
+void party_array() {
+
+    int boys[BOYS_SIZE] = { 0 };
+    int girls[GIRLS_SIZE] = { 0 };
+    int result[BOYS_SIZE + GIRLS_SIZE] = { 0 };
+
+    for (size_t i = 0; i < BOYS_SIZE; ) {
+		// generating one boy height
+        size_t height = rand() % 50 + 150;
+		// if it is even we add it to the array by keeping the array sorted
+        if (height % 2 == 1) {
+            insert_elem(boys, i, height);
+            i++;
+        }
+    }
+    for (size_t i = 0; i < GIRLS_SIZE; ) {
+        size_t height = rand() % 50 + 150;
+		// same as boys but even instead of odd
+        if (height % 2 == 0) {
+            insert_elem(girls, i, height);
+            i++;
+        }
+    }
+
+	std::cout << "Boys array: \n";
+	print_array(boys, BOYS_SIZE);
+	std::cout << "\nGirls array: \n";
+	print_array(girls, GIRLS_SIZE);
+
+    size_t cnt = 0;
+    for (size_t i = BOYS_SIZE; i > 0; i--) {
+
+		// get the two smallest people and 'remove' them so
+		// that they dont take part in the next minimum
+        int smallest_boy = get_min(boys, i);
+        int smallest_girl = get_min(girls, i);
+        result[cnt++] = smallest_boy;
+        result[cnt++] = smallest_girl;
+    }
+
+	std::cout << "\nComibned array: \n";
+    print_array(result, BOYS_SIZE + GIRLS_SIZE);
+}
+
+void print_matrix(int matrix[][MAX_COLS], size_t rows)
+{
```

Comment:
where is the bracket? And the output can be formatted a just a little bit better. We've got code to do so. The idea is the use std::setw